RL.py

```diff
@@ -0,0 +1,274 @@
+import sys
+import math
+import time
+import numpy as np
+import tensorflow as tf
+from pyomo.dae import *
+from pyomo.environ import *
+from tensorflow.keras import layers, regularizers
+from CACTO import CACTO
+
+
+class RL_AC(CACTO):
```

Comment:
Having both RL_AC and TO as child classes of CACTO is potentially problematic because each of them will have its own version of the actor and the critic network, which means that the actor network used to initialize TO is not the same network being updated by RL_AC. Now this problem is not affecting the code because you're using `CACTO.` instead of `self.` to access the networks, but it's not a clean solution in my opinion. You should switch to using `self.`, make `TO` independent of `CACTO`, and move the code to compute the initial guess for TO outside of the TO class.